### ISSUE TITLE
Transfer Learning Model

### ISSUE THREAD

**Issue by Yuliy97** [JKChenFZ/Minerva#69]
## Why/What
- Use transfer learning to create a keras model that takes in images and output two categories(distressed or nondistressed)

## Guideline
1 . The model should be trained and fine tune on FER-13
2 . The baseline model should be trained on a different dataset.

**Comment by Yuliy97** [comment]
File is too big probably not going to be push into the repo

**Comment by Yuliy97** [comment]
Updates data to generate pixelated face images to mimic low latency.

**Comment by Yuliy97** [comment]
Four different Transfer Learning Models:
- ImageNet Xception
- ImageNet ResNet
- ImageNet VGG
- FacialException ResNet

All models are trained via hyper parameter optimization with Adams, SGD, RMSprop optimizer. Models are trained with early stopping when the validation loss no longer decreases for 3 epochs and reduce learning rate on plateau when the validation loss plateau for 2 epochs.

**Comment by Yuliy97** [comment]
- ImageNet Xception
   - Adams:     Accuracy: 0.71, Recall: 0.70,  Presicion:  0.65
   - SGD:     Accuracy: 0.60, Recall: 0.56,  Presicion: 0.52
   - RMSprop:     Accuracy: 0.72, Recall: 0.70,  Presicion: 0.65


- ImageNet ResNet50
   - Adams:     Accuracy: 0.81, Recall: 0.75,  Presicion:  0.74
   - SGD:     Accuracy: 0.76, Recall: 0.74,  Presicion: 0.71
   - RMSprop:     Accuracy: 0.81, Recall: 0.78,  Presicion: 0.74


- ImageNet VGG19
   - Adams:     Accuracy: 0.82, Recall: 0.79,  Presicion:  0.78
   - SGD:     Accuracy: 0.77, Recall: 0.75,  Presicion: 0.71
   - RMSprop:     Accuracy: .82, Recall: 0.78,  Presicion: 0.79


- FacialException ResNet
   - Adams:     Accuracy: 0.61, Recall: 0.61,  Presicion:  0.53
   - SGD:     Accuracy: 0.57, Recall: 0.60,  Presicion: 0.49
   - RMSprop:     Accuracy: 0.57, Recall: 0.58,  Presicion: 0.49

**Comment by Jason-KChen** [comment]
Actually, before we close this task. It is in your best interest to commit your colab/jupyter notebooks. You can commit them under `extension` folder with your custom sub-folder. I have done so in `backend/py_scripts` as an example. I would advise again putting them in my folder since the dependencies are completely different. @Yuliy97
